### PR TITLE
DES Heli Mission Fix

### PR DESCRIPTION
**Pull request by PoweredByPot**
The mission is working as well as any other does at this point. There are a few things that could be improved:

- Right now only a two man sentry team is sent, and they don't get out of the vehicle properly when they arrive at the waypoint. The mission could use adequate defenses, maybe an armed escort?
- Originally the vehicles were loaded on a transport HEMTT (lol), I replaced the GOVT side with a flatbed, but we don't define one for INVADERS yet.
- The actual load position is on top and slightly behind center, when using a flatbed, the position could be adjusted to be on the bed properly.
- Right now we allow ANY transport chopper, which can be quite ridiculous when a Chinook is on top. Only the Little Bird fits and looks appropriate.

### CODE REVIEW

**Comment by Toakan** [comment]
You're trying to merge this to Master @PoweredByPot  :D

File: A3-Antistasi/Missions/DES_Heli.sqf [hunk 1]
```diff
@@ -34,46 +30,38 @@ while {true} do
...
+typeVehX = if (_sideX == Occupants) then {selectRandom (vehNATOTransportHelis + vehNATOAttackHelis)} else {selectRandom (vehCSATAttackHelis + vehCSATTransportHelis)};
 
-_posCrashMrk = [_poscrash,random 500,random 360] call BIS_fnc_relPos;
 _posCrash = _posCrashOrig findEmptyPosition [0,100,_typeVehX];
+
 if (count _posCrash == 0) then
 	{
 	if (!isMultiplayer) then {{ _x hideObject true } foreach (nearestTerrainObjects [_posCrashOrig,["tree","bush"],50])} else {{[_x,true] remoteExec ["hideObjectGlobal",2]} foreach (nearestTerrainObjects [_posCrashOrig,["tree","bush"],50])};
-	_posCrash = _posCrashOrig;
+	_posCrash = _posCrashOrig; //why find a valid location just to overwrite it?
```
Comment (wurzel0701) [inline A3-Antistasi/Missions/DES_Heli.sqf]: If no empty position is found in line 35, it removed all terrain objects around it to create a flat surface and just spawns it on the spot. This is only overwritten if lin 35 does not returns a result

File: A3-Antistasi/Missions/DES_Heli.sqf [hunk 2]
```diff
@@ -34,46 +30,38 @@ while {true} do
...
-_heli = createVehicle [_typeVehX, _poscrash, [], 0, "CAN_COLLIDE"];
+_effect = createVehicle ["CraterLong", _posCrash, [], 0, "CAN_COLLIDE"];
+_heli = createVehicle [_typeVehX, _posCrash, [], 0, "CAN_COLLIDE"];
 _heli attachTo [_effect,[0,0,1.5]];
-_smokeX = "test_EmptyObjectForSmoke" createVehicle _poscrash; _smokeX attachTo[_heli,[0,1.5,-1]];
+_smokeX = "test_EmptyObjectForSmoke" createVehicle _posCrash; _smokeX attachTo [_heli,[0,1.5,-1]];
 _heli setDamage 0.9;
 _heli lock 2;
-_vehiclesX = _vehiclesX + [_heli,_effect];
+_vehiclesX append [_heli,_effect];
```
Comment (wurzel0701) [inline A3-Antistasi/Missions/DES_Heli.sqf]: pushBack is a little bit faster, would be nice to use this

File: A3-Antistasi/Missions/DES_Heli.sqf [hunk 3]
```diff
@@ -86,35 +74,36 @@ while {true} do
...
-_vehicle=[position _road, 0,_typeVehX, _sideX] call bis_fnc_spawnvehicle;
+_vehicle = [position _road, 0,_typeVehX, _sideX] call bis_fnc_spawnvehicle;
 _veh = _vehicle select 0;
 [_veh] call A3A_fnc_AIVEHinit;
 //[_veh,"Escort"] spawn A3A_fnc_inmuneConvoy;
 _vehCrew = _vehicle select 1;
 {[_x] call A3A_fnc_NATOinit} forEach _vehCrew;
 _groupVeh = _vehicle select 2;
-_soldiers = _soldiers + _vehCrew;
+_soldiers append _vehCrew;
```
Comment (wurzel0701) [inline A3-Antistasi/Missions/DES_Heli.sqf]: Same here

**Comment by wurzel0701** [comment]
There is an open case to handle, the truck loads the helicopter and drives back, what if I steal it on the way back and bring it the HQ and garage it there. I should have won the mission in this case, but the real outcome would be a mission failed I guess

File: A3-Antistasi/Missions/DES_Heli.sqf (hunk 2 above)
Comment (PoweredByPot) [reply A3-Antistasi/Missions/DES_Heli.sqf]: pushBack is only for a single element, append combines two arrays only.

File: A3-Antistasi/Missions/DES_Heli.sqf (hunk 3 above)
Comment (PoweredByPot) [reply A3-Antistasi/Missions/DES_Heli.sqf]: array, not an element

File: A3-Antistasi/Missions/DES_Heli.sqf (hunk 1 above)
Comment (PoweredByPot) [reply A3-Antistasi/Missions/DES_Heli.sqf]: removed, for some reason I had it in my head that counting arrays starts at 0 like indexes.

**Comment by PoweredByPot** [comment]
> There is an open case to handle, the truck loads the helicopter and drives back, what if I steal it on the way back and bring it the HQ and garage it there. I should have won the mission in this case, but the real outcome would be a mission failed I guess

The heli is locked when it spawns, preventing players from stealing it. The Transport Truck AI's waypoint settings should prevent it from stopping unless destroyed.

**Comment by wurzel0701** [comment]
> The heli is locked when it spawns, preventing players from stealing it. The Transport Truck AI's waypoint settings should prevent it from stopping unless destroyed.

Still, I shoot the driver, the vehicle stops (as no one drives) and then bring back the truck with the heli attached to it to the HQ and garage the heli, I don't have to move into the heli for this

**Comment by Spoffy** [comment]
I've merged this in, but Wurzel's right. If you could open a separate PR with the fix, that'd be great.

**Comment by Spoffy** [comment]
Closed as merged, new issue opened for fix (#307)

**Pull request closed without merging**